addons/mail/static/src/core/common/store_service.js

```diff
@@ -386,30 +387,38 @@ export class Store extends BaseStore {
             ev.preventDefault();
             this.openChat({ partnerId: id });
             return true;
-        } else if (link.classList.contains("o_message_redirect_transformed") && id) {
+        } else if (link.classList.contains("o_message_redirect")) {
             const message = this["mail.message"].get(id);
             const targetThread = message?.thread;
+            const sendError = () =>
+                this.env.services.notification.add(_t("This conversation isn’t available."), {
+                    type: "danger",
+                });
             if (targetThread) {
                 targetThread.checkReadAccess().then((hasAccess) => {
                     if (hasAccess) {
                         targetThread.highlightMessage = message;
-                        const wasOpen = targetThread.open({ focus: true });
-                        if (!wasOpen) {
+                        let isOpen = targetThread.eq(thread);
+                        if (!isOpen) {
+                            isOpen = targetThread.open({ focus: true, swapOpened: false });
+                        }
+                        if (!isOpen) {
                             window.open(link.href);
                         }
                     } else {
                         if (this.self_partner) {
-                            this.env.services.notification.add(
-                                _t("You do not have the permission to access this thread."),
-                                { type: "warning" }
-                            );
+                            sendError();
                         } else {
                             window.open(link.href);
                         }
                     }
                 });
                 ev.preventDefault();
                 return true;
+            } else if (link.getAttribute("href")?.startsWith(getOrigin())) {
+                sendError();
+                ev.preventDefault();
```

Comment:
I don't get why we just don't use `preventDefault` at the beginning of the function?

Reply:
All cases that we don't handle have to keep the default behavior.